module_utils/network_lsr/argument_validator.py

```diff
@@ -1064,6 +1064,13 @@ def _validate_post_state(self, value, name, result):
         """
         actions = []
         state = result.get("state")
+        persistent_state = result.get("persistent_state")
+        if persistent_state and state in self.VALID_PERSISTENT_STATES:
+            raise ValidationError(
+                name,
+                "State cannot be '{0}' if persistent_state is specified".format(state),
+            )
+
         if state in self.VALID_PERSISTENT_STATES:
             del result["state"]
             persistent_state_default = state
```

Comment:
```suggestion
        persistent_state = result.get("persistent_state")
        if state in self.VALID_PERSISTENT_STATES:
            if persistent_state:
                raise ValidationError(
￼                name,
￼                "State cannot be '{0}' if persistent_state is specified".format(state),
￼            )
            
            del result["state"]
```

What do you think? Everything until `# default persistent_state to present (not done via default_value in the` can probably be removed.

Reply:
@tyll: I also think this makes much more sense now. Anyway, I think is better to just set state to None after giving its value to persistent state (instead of using `del result["state"]`), since the `result` dict is modified in lines 1103:1105 with the rest of the properties, improving readability. (You can see it on the updated commit)